trie/secure_trie.go

```diff
@@ -24,6 +24,16 @@ import (
 	"github.com/ethereum/go-ethereum/triedb/database"
 )
 
+// preimageStore wraps the methods of a backing store for reading and writing
+// trie node preimages.
+type preimageStore interface {
+	// Preimage retrieves the preimage of the specified hash.
+	Preimage(hash common.Hash) []byte
+
+	// InsertPreimage commits a set of preimages along with their hashes.
+	InsertPreimage(preimages map[common.Hash][]byte)
```

Comment:
Maybe `get` and `put` or `insert` or something. `tr.preimages.Preimage()` and `tr.preimages.InsertPreimage()` feel kinda cumbersome

Reply:
unfortunately this two methods are implemented by the `triedb`. `Get` or `Put` is too generic for triedb. 
I can try to improve the interface implementation a bit in a following PR.